### ISSUE TITLE
Remove Scale Along Z property in collections

### ISSUE THREAD

**Issue by AGulev** [defold/defold#9529]
**Task Overview:**

The goal is to remove the old hidden `scale_along_z` field from collections in the engine, which forces the engine to ignore the Z-component of scale for all entities (Game Objects and components). Since removing it outright would break existing projects that rely on this behavior, the objective is to gradually phase out this feature while providing an alternative mechanism for developers to control the scaling behavior.

**Proposed Steps:**

1. **Major Engine Release with Breaking Change**  
   The engine's next major release will introduce a breaking change by removing the `scale_along_z` behavior. This will ensure that future versions of the engine no longer rely on hidden fields that manage scaling, improving maintainability.

2. **Introduce a New `scale.xy` Property in the Lua API**  
   To replace the old behavior, a new property (`scale.xy` or a similar name) will be introduced for Game Objects and components via the Lua API. This new property will specifically ignore the Z-component of the scale. 

   - **For Old Projects:** Projects relying on the old behavior will need to replace all instances of `scale` properties (used in `go.set()` and `go.animate()`) with this new `scale.xy` property. 
   - This migration will be similar to the earlier transition from `gui.PROP_ROTATION` to `gui.PROP_EULER`, making it easier for developers to update their projects without thoroughly inspecting all scale values.

3. **Editor Migration Tool**  
   To assist developers with this transition, an editor migration tool will be introduced. This tool will:
   - Automatically replace all instances of `scale.z` in Game Objects and components within collections with `scale_along_z:0` to 1.0 or a similar mechanism to remove reliance on the deprecated field.
   - This automated tool will simplify the migration process, helping developers update their projects without manual intervention.

**Reasoning for the New `scale.xy` Property:**  
The new property allows developers to easily replace constants in their code, streamlining the process. Replacing `scale` with `scale.xy` across the project is far simpler than manually investigating and adjusting individual values for every use of `scale` property, making the transition smoother and less error-prone.

**Expected Outcome (REQUIRED):**
The engine will no longer contain any references or code related to `scale_along_z`. Instead, the correct scaling behavior (whether 2D or 3D) will be explicitly managed by the user through the Lua API (`scale.xy`), ensuring that scaling logic is clear and intentional in all projects. This will improve maintainability, remove legacy dependencies, and allow users greater control over their scaling behavior.

**Comment by AGulev** [comment]
Related: 
https://github.com/defold/defold/issues/7441
https://github.com/defold/defold/issues/6165
https://github.com/defold/defold/issues/5679

**Comment by britzl** [comment]
> 2\. `2d_scale` or a similar name

I vote for `scale2d`

**Comment by AGulev** [comment]
> I vote for `scale2d`

Replaced with `scale.xy`

**Comment by aglitchman** [comment]
It would be really cool if you could temporarily add at least just displaying the value of ‘scale_along_z’ in the collection. 

Because I spent a working day searching for a bug because I forgot to change 0 to 1 in a newly created collection. It was a real pain 😭

**Comment by seel1ch** [comment]
Yes, some checkbox would be helpful. I also recently forgot about this setting in a new project 😅
